test/unit/test-amp-story-player.js

```diff
@@ -522,6 +526,33 @@ describes.realWin('AmpStoryPlayer', {amp: false}, (env) => {
       expect(readySpy).to.have.been.calledOnce;
     });
 
+    it('get page attachment state should send message', async () => {
+      buildStoryPlayer();
+      await manager.loadPlayers();
+
+      await playerEl.getStoryState('page-attachment');
+
+      messagingMock
+      .expects('sendRequest');
+
+      messagingMock
+        .expects('sendRequest')
+        .withArgs('getDocumentState', {state: 'PAGE_ATTACHMENT_STATE'});
+    });
+
+    it('close page attachment means button is displayed', async () => {
```

Comment:
can we create another test where the page attachment is open? we can do this by sending a fake message from the story like here https://github.com/ampproject/amphtml/blob/690b831ba8e014cfbf1d0b1d7910608f49e76838/test/unit/test-amp-story-player.js#L73

Reply:
So I wrote a swipeUp and swipeDown, but I'm not sure if that's correct.
```
 function swipeUp() {
    const touchStartEvent = {touches: [{screenX: 150, screenY: 200}]};
    fireHandler['touchstart']('touchstart', touchStartEvent);

    const touchMove = {touches: [{screenX: 150, screenY: 100}]};
    fireHandler['touchmove']('touchmove', touchMove);

    const touchEndEvent = {touches: [{screenX: 150, screenY: 100}]};
    fireHandler['touchend']('touchend', touchEndEvent);
  }
```

Does that look like it should work?

Reply:
The `messaging` methods associated with `touchstart`, `touchmove` and `touchend` currently don't consider vertical swipes.

Reply:
Sorry, I meant more something like:

```
const fakeEvent = {state:'PAGE_ATTACHMENT_STATE', value: 'true'}
fireHandler['documentStateUpdate'](fakeEvent)

```

Reply:
Ohh, gotcha, will try that out!